manifests/init.pp

```diff
@@ -380,6 +381,7 @@
   String $service_name = 'patroni',
   String $service_ensure = 'running',
   Boolean $service_enable = true,
+  Optional[String] $custom_pip_provider = undef,
```

Comment:
I think the pip_provider is always a path, in which case it should have the type `Stdlib::Absolutepath`

Reply:
It can be relative, the default is `pip3`, it's just the executable to run to execute pip which can be relative if in PATH which often it is.

Reply:
https://github.com/voxpupuli/puppet-python/blob/master/manifests/pip.pp#L54